tools/replay/desktop_main.cpp

```diff
@@ -88,29 +89,82 @@ void WaitForExit() {}
 
 const char kLayerEnvVar[] = "VK_INSTANCE_LAYERS";
 
-#if defined(D3D12_SUPPORT)
-bool BrowseFile(const std::string&                           input_filename,
-                const gfxrecon::decode::DumpResourcesTarget& dump_resources_target,
-                gfxrecon::decode::TrackDumpDrawCall&         out_track_dump_target)
+void RunVulkanPreProcessConsumer(const std::string&                      input_filename,
+                                 gfxrecon::decode::VulkanReplayOptions&  replay_options,
+                                 gfxrecon::decode::VulkanReplayConsumer& replay_consumer)
 {
-    gfxrecon::decode::TrackDumpDrawCall* track_dump_target = nullptr;
+    gfxrecon::decode::FileProcessor file_processor;
+    if (file_processor.Initialize(input_filename))
+    {
+        gfxrecon::decode::VulkanPreProcessConsumer pre_process_consumer;
+
+        bool process_dump_resources =
+            replay_options.enable_dump_resources && replay_options.dump_resources_block_indices.empty();
+        if (process_dump_resources)
+        {
+            pre_process_consumer.EnableDumpResources(replay_options.dump_resources_target);
+        }
+
+        gfxrecon::decode::VulkanDecoder decoder;
+        decoder.AddConsumer(&pre_process_consumer);
+        file_processor.AddDecoder(&decoder);
+        file_processor.ProcessAllFrames();
+
+        replay_options.enable_vulkan = pre_process_consumer.WasVulkanAPIDetected();
+
+        if (replay_options.enable_vulkan)
+        {
+            if (process_dump_resources)
+            {
+                replay_options.dump_resources_block_indices = pre_process_consumer.GetDumpResourcesBlockIndices();
+            }
+
+            if (replay_options.enable_dump_resources)
+            {
+                // Process --dump-resources-block-indices arg.
+                if (!gfxrecon::parse_dump_resources::parse_dump_resources_arg(replay_options))
+                {
+                    GFXRECON_LOG_FATAL("There was an error while parsing dump resources indices. Terminating.");
+                    return;
```

Comment:
GFXRECON_LOG_FATAL prints an error message and returns, then this method will return, I think to continue trying to do the replay. Should the return be changed to a call to exit()?

Reply:
Added exit(0);